### PR TITLE
Port the SpirvLowerConstImmediateStore and SpirvLowerMath passes to the new pass manager

### PR DESCRIPTION
**Pull request by bsaleil**

### CODE REVIEW

File: llpc/lower/llpcSpirvLowerMath.cpp [hunk 1]
```diff
@@ -220,18 +202,42 @@ void SpirvLowerMath::disableFastMath(Value *value) {
 
 #define DEBUG_TYPE DEBUG_TYPE_CONST_FOLDING
 
+// =====================================================================================================================
+// Executes constand folding SPIR-V lowering pass on the specified LLVM module.
```
Comment (piotrAMD) [inline llpc/lower/llpcSpirvLowerMath.cpp:206]: constand -> constant

File: llpc/lower/llpcSpirvLowerMath.cpp [hunk 2]
```diff
@@ -220,18 +202,42 @@ void SpirvLowerMath::disableFastMath(Value *value) {
...
+//
+// @param [in/out] module : LLVM module to be run on (empty on entry)
+bool LegacySpirvLowerMathConstFolding::runOnModule(Module &module) {
+  return Impl.runImpl(module, [&]() -> TargetLibraryInfo & {
+    return getAnalysis<TargetLibraryInfoWrapperPass>().getTLI(*(Impl.getEntryPoint()));
+  });
+}
+
+// =====================================================================================================================
+// Executes constand folding SPIR-V lowering pass on the specified LLVM module.
```
Comment (piotrAMD) [inline llpc/lower/llpcSpirvLowerMath.cpp:216]: constand -> constant

**Review by piotrAMD — COMMENTED** [review]
LGTM modulo the typos.

**Comment by bsaleil** [comment]
Fixed typo in comments.

**Review by amdrexu — APPROVED** [review]

**Comment by JaxLinAMD** [comment]
retest this please ( token to trigger test)

**Comment by amdvlk-admin** [comment]
### Test summary for commit e4e1d9064c3f6d0aae1afb1c694086d3cdda3296
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev cd4f8b28b5d923e8d65a1ef1d3dee5137a89d6bd</li>
<li>SPVGEN: dev 8705e87a8b4aaed55a89f79ed423612ca310bfdd</li>
<li>XGL: dev eecbf8b682753c266cac6b2d1f3032733e5d2049</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 68e5420f0ba94be9cae6fbac848d746592d03f42</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/221024)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 31062/55256 (56.2%)</li>
<li>Failed: 0/55256 (0.0%)</li>
<li>Not Supported: 24194/55256 (43.8%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 31062/55256 (56.2%)</li>
<li>Failed: 0/55256 (0.0%)</li>
<li>Not Supported: 24194/55256 (43.8%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 31436/55256 (56.9%)</li>
<li>Failed: 0/55256 (0.0%)</li>
<li>Not Supported: 23820/55256 (43.1%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 34187/55256 (61.9%)</li>
<li>Failed: 0/55256 (0.0%)</li>
<li>Not Supported: 21069/55256 (38.1%)</li>
</ul></details>


</ul></details>

**Review by piotrAMD — APPROVED** [review]

**Pull request merged**